src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int capacity;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] array;
 
     @Override
     public void put(K key, V value) {
-
+        if (capacity == 0) {
+            array = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+            capacity = INITIAL_CAPACITY;
+            threshold = (int) (capacity * LOAD_FACTOR);
+        }
+        if (size + 1 > threshold) {
+            resize();
+        }
+        int hash = (key == null) ? 0 : Math.abs(key.hashCode());
+        int bucketIndex = hash % capacity;
+        Node<K, V> node = array[bucketIndex];
+        if (node == null) {
+            array[bucketIndex] = new Node<>(hash, key, value, null);
+            size = size + 1;
+            return;
+        }
+        if (key == node.key || (key != null && key.equals(node.key))) {
+            node.value = value;
+            return;
+        }
+        while (node.next != null) {
+            node = node.next;
+            if (key == node.key || (key != null && key.equals(node.key))) {
+                node.value = value;
+                return;
+            }
+        }
+        node.next = new Node<>(hash, key, value, null);
+        size = size + 1;
     }
 
     @Override
     public V getValue(K key) {
+        if (capacity == 0) {
+            return null;
+        }
```

Comment:
Do we need this ?

Reply:
yes
this one-step check saves us from unnecessary iteration and division by zero)

Reply:
will capacity ever be 0? (what is the difference between size and capacity?)
also, I think this check is redundant, we won't iterate in the loop if the node is null

Reply:
One of the tests tries to get an element before putting anything. 
In my solution, the array of nodes is only created when adding first element, thus division by 0 may occur ( % capacity) left unchecked